cherrypy/lib/auth_digest.py

```diff
@@ -38,11 +44,39 @@ def md5_hex(s):
     return md5(ntob(s, 'utf-8')).hexdigest()
 
 
+def sha256_hex(s):
+    """Return hexdigest of sha256sum."""
+    return sha256(ntob(s, 'utf-8')).hexdigest()
+
+
+def sha512_hex(s):
+    """Return hexdigest of sha512sum."""
+    return sha512(ntob(s, 'utf-8')).hexdigest()
+
+
+def choose_hexdigest_method(algorithm):
```

Comment:
This function could just return a callable, one of `md5()`, `sha256()` or `sha512()`. And the consumer would invoke it in a uniform way.